src/utilities/telegram_utils.rs

```diff
@@ -22,9 +21,48 @@ impl MainUsername for User {
     }
 }
 
-pub struct MessageImage {
-    pub file: File,
-    pub mime_type: Cow<'static, str>,
+pub enum MessageAttachment {
+    Photo(Photo),
+    Video(Video),
+    Animation(Animation),
+    Audio(Audio),
+    Voice(VoiceNote),
+    Document(Document),
+}
+
+impl MessageAttachment {
+    pub fn filesize(&self) -> i64 {
+        match self {
+            MessageAttachment::Photo(photo) => largest_photo(photo).map(|file| file.size).unwrap_or(0),
+            MessageAttachment::Video(video) => video.video.size,
+            MessageAttachment::Animation(animation) => animation.animation.size,
+            MessageAttachment::Audio(audio) => audio.audio.size,
+            MessageAttachment::Voice(voice) => voice.voice.size,
+            MessageAttachment::Document(document) => document.document.size,
+        }
+    }
+
+    pub fn file_id(&self) -> i32 {
+        match self {
+            MessageAttachment::Photo(photo) => largest_photo(photo).map(|file| file.id).unwrap_or(0),
+            MessageAttachment::Video(video) => video.video.id,
+            MessageAttachment::Animation(animation) => animation.animation.id,
+            MessageAttachment::Audio(audio) => audio.audio.id,
+            MessageAttachment::Voice(voice) => voice.voice.id,
+            MessageAttachment::Document(document) => document.document.id,
+        }
+    }
+
+    pub fn mime_type(&self) -> Cow<'static, str> {
+        match self {
+            MessageAttachment::Photo(_) => Cow::Borrowed("image/jpeg"),
+            MessageAttachment::Video(video) => Cow::Owned(video.mime_type.clone()),
+            MessageAttachment::Animation(_) => Cow::Borrowed("video/mp4"),
```

Comment:
the animation may also be a GIF. read the `animation.animation.mime_type` property